### PR TITLE
SSE Live Reload

### PR DESCRIPTION
**Pull request by tcardlab**
In regard to #6. 

Feel free to ignore, just sharing incase others are interested.

### CODE REVIEW

File: src/dev-server.ts [hunk 1]
```diff
@@ -89,14 +115,21 @@ class Server {
 
       const found = (
         getFile(url) ??
+        getFile(url + '.html') ??
```
Comment (sdegutis) [inline src/dev-server.ts:118]: Why?

File: src/dev-server.ts [hunk 2]
```diff
@@ -89,14 +115,21 @@ class Server {
...
         getFile(path.posix.join(url, 'index.html'))
       );
 
       if (found) {
         res.statusCode = 200;
         const contentType = mimetypes.contentType(path.extname(found.url));
         res.setHeader('content-type', contentType || 'application/octet-stream');
-        res.end(found.blob);
+        if (process.env.NODE_ENV !== 'production' && contentType.startsWith('text/html')) {
```
Comment (sdegutis) [inline src/dev-server.ts]: Is NODE_ENV ever production? This is a dev server only. Usually it just generates static files for prod.

File: src/dev-server.ts [hunk 3]
```diff
@@ -50,17 +51,42 @@ export function startDevServer(runtime: Runtime, config?: { port?: number }) {
     .on('unlink', pathUpdated));
 }
 
+function __hot_reload () {
+  const eventSource = new EventSource('/hot-reload');
```
Comment (sdegutis) [inline src/dev-server.ts:55]: I've never heard of this before.

**Comment by sdegutis** [comment]
Honestly I wanna try this now.

@tcardlab if I were to merge this, I'd need a really simple contribution form signed, to ensure that your code being merged into my project doesn't block me in the future.

File: src/dev-server.ts (hunk 1 above)
Comment (tcardlab) [reply src/dev-server.ts:118]: ✨aesthetics✨ and the convenience of 5 saved characters
something.com/test.html => something.com/test

Comment (sdegutis) [reply src/dev-server.ts:118]: @tcardlab I've personally never liked how https://github.com/sdegutis and https://github.com/sdegutis/ are the same page but technically two different URLs, and there's no redirect. Maybe it's just me...

Comment (tcardlab) [reply src/dev-server.ts:118]: 100% agree that ones annoying, but cant recall the last time I saw .html a url in the past month.
I wasn't really anticipating a merge so I was being a bit opinionated here

File: src/dev-server.ts (hunk 2 above)
Comment (tcardlab) [reply src/dev-server.ts]: yea, that was dumb...
pretend you didn't see that lol

**Comment by tcardlab** [comment]
> Honestly I wanna try this now.
> 
> @tcardlab if I were to merge this, I'd need a really simple contribution form signed, to ensure that your code being merged into my project doesn't block me in the future.

yea sure

**Pull request closed without merging**